zofm00-sp2

Zofak opened the pull request:
https://eso.vse.cz/~zofm00/sp2/

Zofak committed:
zofm00-sp2

https://eso.vse.cz/~zofm00/sp2/

Zofak committed:
V2

Version ready for exam

nvbach91 commented on www/zofm00/sp2/index.html:
```diff
@@ -0,0 +1,117 @@
+            <button id="findName">Vyhledat</button>
+            <div id="outputName"></div>
+            <div class="descriptionDate">Datum</div>
+            <select id="month" onchange="ChangeDays()">
```
bez inline JS

nvbach91 commented on www/zofm00/sp2/index.html:
```diff
@@ -0,0 +1,117 @@
+    </header>
+    <main>
+        <h4>Zadejte jméno nebo datum do příslušné kolonky</h4>
+        <div class=find>
```
nevalidni HTML

nvbach91 replied:
tady pouzivejte form

nvbach91 commented on www/zofm00/sp2/index.html:
```diff
@@ -0,0 +1,117 @@
+            
+
+        <div class="bottom">
+                <h4>Má uložená jména</h4>
```
naformatovat kod

nvbach91 commented on www/zofm00/sp2/index.html:
```diff
@@ -0,0 +1,117 @@
+        <div>Semestrální práce pro potřeby 4IZ268, Autor: Martin Zof, 2021</div>
+    </footer>
+
+    <script src="js/jquery-3.5.1.min.js"></script>
```
jquery naimportovat z CDN

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+days['nov'] = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30'];
+days['dec'] = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'];
+
+function ChangeDays() {
```
nazvy funkci v camelCase

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+}
+
+function LoadSavedNames() {
+    const name1 = $('#saved1');
```
udelejte to tak aby se dalo ukladat neomezene mnozstvi jmen

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+        var inputName = $('#name').val(); //čtení vstupu
+
+
+        if (inputName.length == 0) {//hlídá, zda kolonka není prázdná
```
tohle jako osetreni uzivatelskeho vstupu nestaci, provedte dalsi transformace stringu

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+    }
+
+    function SaveName(name, date) {
+        const name1 = $('#saved1');
```
provedte kontrolu existujicich jmen v seznamu ulozenych

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+            try {
+                var replyField = resp.split(";");
+                var today = $('#today');
+                if (replyField[0] === '0101') {
```
informace ohledne svatku ulozke do objektu podle toho vypiste ty hlasky

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+
+    $.ajax({
+        method: 'GET',
+        url: 'https://svatky.adresa.info/txt',
```
pouzivejte json format odpovedi

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+                var replyField = resp.split(";");
+                var today = $('#today');
+                if (replyField[0] === '0101') {
+                    today.text('Dnes je Nový rok a Den obnovy samostatného českého státu.');
```
informace ohledne svatku ulozke do objektu podle toho vypiste ty hlasky tady i u vyhledani datumu

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+                method: 'GET',
+                url: 'https://svatky.adresa.info/txt?date=' + daySearch + monthSearch,
+                success: (resp) => {
+                    try {
```
proc je tady try catch?

nvbach91 commented on www/zofm00/sp2/js/main.js:
```diff
@@ -0,0 +1,468 @@
+                    }
+                },
+                error: (error) => { console.log(error); },
+                complete: LoaderHide()
```
pouzivejte promise

Zofak committed:
Oprava v rámci obhajoby

Opraveny názvy funkcí
Vyřešena validace vstupu
Přidán form 
Opraveno nevalidní html
Odejmut inline js
Kód naformátován
JQuery z CDN
Lze uložit více jmen
Nelze uložit stejné jméno vícekrát
Aplikace již využívá JSON, nikoli text
Přidány objekty namísto else if a switch

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
